servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WireLoggingInitializer.java

```diff
@@ -15,36 +15,30 @@
  */
 package io.servicetalk.transport.netty.internal;
 
-import io.netty.channel.Channel;
-import io.netty.handler.logging.LogLevel;
-import io.netty.handler.logging.LoggingHandler;
-
-import javax.annotation.Nullable;
+import io.servicetalk.logging.api.FixedLevelLogger;
 
-import static io.servicetalk.transport.netty.internal.NettyLoggerUtils.getNettyLogLevel;
+import io.netty.channel.Channel;
 
 /**
  * A {@link ChannelInitializer} that enables wire-logging for all channels.
  * All wire events will be logged at trace level.
  */
 public class WireLoggingInitializer implements ChannelInitializer {
-    @Nullable
-    private final LoggingHandler loggingHandler;
+    private final ServiceTalkWireLogger loggingHandler;
 
     /**
-     * Create an instance that logs at trace level.
+     * Create an instance.
      *
-     * @param loggerName The name of the logger to log wire events.
+     * @param logger The logger to use for log wire events.
+     * @param logUserData {@code true} to log user data. {@code false} to not log user data.
      */
-    public WireLoggingInitializer(final String loggerName) {
-        LogLevel logLevel = getNettyLogLevel(loggerName);
-        loggingHandler = logLevel != null ? new LoggingHandler(loggerName, logLevel) : null;
+    public WireLoggingInitializer(final FixedLevelLogger logger,
+                                  final boolean logUserData) {
+        loggingHandler = new ServiceTalkWireLogger(logger, logUserData);
```

Comment:
Instead of introducing a custom `ServiceTalkWireLogger` can we use `LoggingHandler(name, level, logUserData ? ByteBufFormat.HEX_DUMP : ByteBufFormat.SIMPLE)` from netty?

Reply:
Netty's `LoggingHandler` will use `toString()` on messages, so depending upon what types are in the pipeline this may log user data (e.g. headers, full http message, etc.).

Reply:
We insert it as the first handler at tcp level where only buffers are expected. If something will change for our tcp initializer in future, we can reconsider approach.

Reply:
latest round of review changes introduces `BoolenSupplier` to dynamically include/exclude data. there is some duplication but I think its simpler just to keep this for now. it is internal so we can consolidate later if we enhance upstream or find a better way.